src/NHibernate/Cache/IBatchableWriteCache.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace NHibernate.Cache
+{
+	/// <summary>
+	/// Defines methods for retrieving and adding multiple objects from/to the cache at once.
+	/// The implementor should use this interface along with <see cref="ICache"/> when the
+	/// cache supports a multiple get and put operation.
+	/// </summary>
+	/// <remarks>
+	/// <threadsafety instance="true" />
+	/// <para>
+	/// All implementations <em>must</em> be threadsafe.
+	/// </para>
+	/// </remarks>
+	public partial interface IBatchableReadWriteCache : IBatchableReadCache
```

Comment:
Are there any reasons to split these interfaces?

Reply:
The only reason is to support caches that have only a batchable read operation like [EnyimMemcached](https://github.com/enyim/EnyimMemcached/blob/develop/Enyim.Caching/IMemcachedClient.cs).